src/neptune/common/backends/utils.py

```diff
@@ -95,14 +95,22 @@ def wrapper(*args, **kwargs):
                 HTTPServiceUnavailable,
                 HTTPGatewayTimeout,
                 HTTPBadGateway,
-                HTTPTooManyRequests,
                 HTTPInternalServerError,
                 NewConnectionError,
                 ChunkedEncodingError,
             ) as e:
                 time.sleep(min(2 ** min(10, retry), MAX_RETRY_TIME))
                 last_exception = e
                 continue
+            except HTTPTooManyRequests as e:
+                response_headers = e.response.json()["headers"]
+                if "retry-after" in response_headers:
+                    retry_after = int(response_headers["retry-after"][0])
+                    time.sleep(min(retry_after, MAX_RETRY_TIME))
+                else:
+                    time.sleep(min(2 ** min(10, retry), MAX_RETRY_TIME))
```

Comment:
`10` is a Magic number -> let's make a constant instead

Reply:
True, it was like that before, but I reformatted it so that it is now always in constant 😇